usr.bin/minimal_count_stats/minimal_count_stats.c

```diff
@@ -116,94 +167,115 @@ main(int argc, char **argv)
 	}
-	// Infer default value for statcounters_format from environment
-	const char* original_fmt = getenv("STATCOUNTERS_FORMAT");
-	if (statcounters_format == (statcounters_fmt_flag_t)-1) {
-		if (original_fmt && strcmp(original_fmt,"csv") == 0) {
-			// If the file already exists, we use CSV_NOHEADER later
-			statcounters_format = CSV_HEADER;
-		}
-		else {
-			statcounters_format = HUMAN_READABLE;
-		}
-	}
 
-	/* Also dump to stderr when -v was passed */
-	// FIXME: should change libstatcounters to have a better API
-	if (verbose) {
-		// Ensure human readable output:
-		unsetenv("STATCOUNTERS_FORMAT");
-		statcounters_dump_with_args(&diff_count, progname, NULL,
-		    architecture, stderr, HUMAN_READABLE);
-		if (original_fmt)
-			setenv("STATCOUNTERS_FORMAT", original_fmt, 1);
-	}
-	FILE* output_file = NULL;
+	FILE *output_file = NULL;
 	if (!output_filename || strcmp(output_filename, "-") == 0) {
 		output_file = stdout;
 	} else {
 		output_file = fopen(output_filename, "a");
-		/* If we are writing to a regular file and the offset is not
-		 * zero omit the CSV header */
+		// Also dump basic stats to stderr when -v was passed
+		if (verbose) {
+			print_basic_human_readable_diff(stderr, &ts_diff,
+			    &diff_count);
+		}
 		if (!output_file) {
-			err(EX_OSERR, "fopen(%s)", output_filename);
+			err(1, "fopen(%s)", output_filename);
 		}
-		if (statcounters_format == CSV_HEADER && ftello(output_file) > 0) {
+		// If we are writing to a regular file and the offset is not
+		// zero omit the CSV header.
+		if (statcounters_format == CSV_HEADER &&
+		    ftello(output_file) > 0) {
 			statcounters_format = CSV_NOHEADER;
 		}
 	}
-	// Unset statcounters format for the dump so that the explicit
-	// argument is used instead of the environment variable
-	// FIXME: change libstatcounters instead of using this hack
-	unsetenv("STATCOUNTERS_FORMAT");
-	statcounters_dump_with_args(&diff_count, progname, NULL,
-	    architecture, output_file, statcounters_format);
-	if (original_fmt)
-		setenv("STATCOUNTERS_FORMAT", original_fmt, 1);
+	if (basic_only || time_only) {
+		if (statcounters_format == HUMAN_READABLE) {
+			print_basic_human_readable_diff(output_file, &ts_diff,
+			    &diff_count);
+		} else {
+			// CSV output requested:
+			// This should be compatible with existing
+			// libstatcounters analysis scripts.
+			if (statcounters_format == CSV_HEADER) {
+				fprintf(output_file,
+				    "progname,archname,cycles,"
+				    "instructions,clock-monotonic\n");
+			} else {
+				assert(statcounters_format == CSV_NOHEADER);
+			}
+			fprintf(output_file, "%s,%s,%jd,%jd,%jd.%09jd\n",
+			    progname, architecture, (intmax_t)diff_count.cycles,
+			    (intmax_t)diff_count.instructions,
+			    (intmax_t)ts_diff.tv_sec,
+			    (intmax_t)ts_diff.tv_nsec);
+		}
+	} else {
+		statcounters_dump_with_args(&diff_count, progname, "",
+		    architecture, output_file, statcounters_format);
+	}
 	exit(WEXITSTATUS(status));
```

Comment:
Because returning from main is too hard? The mind boggles...

Reply:
I wonder where I copied that from...